code/game/objects/items/weapons/twohanded.dm

```diff
@@ -283,20 +283,90 @@
 	w_class = 4
 	slot_flags = SLOT_BACK
 	force_unwielded = 10
-	force_wielded = 18 // Was 13, Buffed - RR
+	force_wielded = 18
 	throwforce = 20
-	throw_speed = 3
+	throw_speed = 4
 	armour_penetration = 10
-	no_spin_thrown = 1 // Thrown spears that spin look dumb. -Fox
+	materials = list(MAT_METAL=1150, MAT_GLASS=2075)
+	hitsound = 'sound/weapons/bladeslice.ogg'
 	attack_verb = list("attacked", "poked", "jabbed", "torn", "gored")
+	sharp = 1
+	edge = 1
+	no_spin_thrown = 1
+	var/obj/item/weapon/grenade/explosive = null
+	var/war_cry = "AAAAARGH!!!"
 
 /obj/item/weapon/twohanded/spear/update_icon()
-	icon_state = "spearglass[wielded]"
-	return
+	if(explosive)
+		icon_state = "spearbomb[wielded]"
+	else
+		icon_state = "spearglass[wielded]"
 
-/obj/item/weapon/twohanded/spear/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
-	playsound(loc, 'sound/weapons/bladeslice.ogg', 50, 1, -1)
-	return ..()
+/obj/item/weapon/twohanded/spear/afterattack(atom/movable/AM, mob/user, proximity)
+	if(!proximity)
+		return
+	if(isturf(AM)) //So you can actually melee with it
+		return
+	if(explosive && wielded)
+		user.say("[war_cry]")
+		explosive.loc = AM

```

Comment:
this should intentinoally set `loc`, otherwise, you're going to have unintended behaviors with some grenades and assemblies